source/beamline/BioXAS/BioXASCarbonFilterFarmActuatorFilterControl.cpp

```diff
@@ -76,7 +76,24 @@ void BioXASCarbonFilterFarmActuatorFilterControl::clearFilters()
 
 void BioXASCarbonFilterFarmActuatorFilterControl::setWindowPreference(double filter, int windowIndex)
 {
-	filterWindowPreferenceMap_.insert(filter, windowIndex);
+	if (indices_.contains(windowIndex)) {
+
+		// Identify any other window indices with the given filter value.
+		// Mark these windows as read-only.
+
+		QList<int> filterMatches = indexFilterMap_.keys(filter);
```

Comment:
Maybe I don't understand what's happening, but why is it the provided preference filter becomes the read only state?